packages/blade/src/components/Tabs/__tests__/Tabs.native.test.tsx

```diff
@@ -0,0 +1,130 @@
+        </TabPanel>
+        <TabPanel value="refunds">
+          <Text>Refunds Panel</Text>
+        </TabPanel>
+      </Tabs>,
+    );
+
+    expect(toJSON()).toMatchSnapshot();
+  });
+
+  it('should render with filled variant', () => {
+    const { toJSON } = renderWithTheme(
+      <Tabs variant="filled">
+        <TabList>
+          <TabItem value="payments">Payments</TabItem>
+          <TabItem value="refunds">Refunds</TabItem>
+        </TabList>
+
+        <TabPanel value="payments">
+          <Text>Payments Panel</Text>
+        </TabPanel>
+        <TabPanel value="refunds">
+          <Text>Refunds Panel</Text>
+        </TabPanel>
+      </Tabs>,
+    );
+
+    expect(toJSON()).toMatchSnapshot();
+  });
+
+  it('should render with isFullWidthTabItem', () => {
+    const { toJSON } = renderWithTheme(
+      <Tabs isFullWidthTabItem>
+        <TabList>
+          <TabItem value="payments">Payments</TabItem>
+          <TabItem value="refunds">Refunds</TabItem>
+        </TabList>
+
+        <TabPanel value="payments">
+          <Text>Payments Panel</Text>
+        </TabPanel>
+        <TabPanel value="refunds">
+          <Text>Refunds Panel</Text>
+        </TabPanel>
+      </Tabs>,
+    );
+
+    expect(toJSON()).toMatchSnapshot();
+  });
+
+  it('should render with large size', () => {
+    const { toJSON } = renderWithTheme(
+      <Tabs size="large">
+        <TabList>
+          <TabItem value="payments">Payments</TabItem>
+          <TabItem value="refunds">Refunds</TabItem>
+        </TabList>
+
+        <TabPanel value="payments">
+          <Text>Payments Panel</Text>
+        </TabPanel>
+        <TabPanel value="refunds">
+          <Text>Refunds Panel</Text>
+        </TabPanel>
+      </Tabs>,
+    );
+
+    expect(toJSON()).toMatchSnapshot();
+  });
+
+  // Skipping because there is some testability issues with react-native-tabview
+  // https://github.com/react-navigation/react-navigation/issues/11493#issuecomment-1777846110
+  it.skip('should switch tabs', () => {
+    const onChangeFn = jest.fn();
+    const { toJSON, getByText, getByRole } = renderWithTheme(
+      <Tabs onChange={(value) => onChangeFn(value)}>
+        <TabList>
+          <TabItem value="payments">Payments Tab</TabItem>
+          <TabItem
+            onClick={() => {
+              console.log('pressed');
+            }}
+            value="refunds"
+          >
+            Refunds Tab
+          </TabItem>
+        </TabList>
+
+        <TabPanel value="payments">
+          <Text>Payments Panel</Text>
+        </TabPanel>
+        <TabPanel value="refunds">
+          <Text>Refunds Panel</Text>
+        </TabPanel>
+      </Tabs>,
+    );
+
+    // Assert initial state
+    expect(getByRole('tab', { name: 'Payments Tab' })).toHaveAccessibilityState({
+      selected: true,
+    });
+    expect(getByText('Payments Panel')).toBeVisible();
+
+    // Click on the second tab
+    fireEvent.press(getByRole('tab', { name: 'Refunds Tab' }));
+
```

Comment:
add panel to be visible assertion here?

Reply:
Can't. testability issues with rn library. 

https://razorpay.slack.com/archives/G01B3LQ9H0W/p1698213664463119